README.md

```diff
@@ -8,43 +8,136 @@ A simple betting game for two players (“Rock-Paper-Scissors”) played inside
-* when both players’ bets are observed on the Head, the gestures chosen are revealed and compared
-    * in case of a tie bets are returned to their respective owners
-    * when there is a winning gesture, all the bets are transferred to the winner
+First, clone the git repository at `https://github.com/mlabs-haskell/hydra-demo.git` and build the application.
+Building may take a while the first time.
 
-At any time each player can decide to close the Head, settling all the payments to Layer 1.
+```
+$ git clone https://github.com/mlabs-haskell/hydra-demo.git
+$ cd hydra-demo
+$ nix build .
+```
 
-## Advantages of Hydra in this use case
+Then pull all the necessary Docker images, and spin-up the devnet. From the repository root:
 
-* Rapid games in close to realtime
-* Lower fees once funds are in the hydra head
-* Micropayment use case
+```
+$ docker-compose pull
+
+$ ./spin-up-devnet-from-scratch
+```
+
+It's worth noting that `spin-up-devnet-from-scratch` invokes `sudo` and may prompt for your password, but fear not! It is there just to remove any stale `./devnet` data directory from previous runs.
+
+Now that the stage is set, we can finally start the application instances.
+
+From the repository root in two separate terminals execute:
+
+```
+nix run . 127.0.0.1 4001 devnet/credentials/alice.sk devnet/protocol-parameters.json
+```
+
+and
+
+```
+nix run . 127.0.0.1 4002 devnet/credentials/bob.sk devnet/protocol-parameters.json
+```
+
+Each of these will spin up an instance of the application, connected to a different hydra-node (cfr. with the `docker-compose.yaml` file). We must also pass in the signing keys for each user, as the CLI will sign txs and submit them to the hydra node.
+## Opening the head
+
+We can now open the head and start playing the RPS game.
+[This](https://hydra.family/head-protocol/core-concepts) page has a good expalnation (alongside nice diagrams) to show all the phases of the Hydra head, an in-depth discussion of those is out of the scope of this tutorial, so we invite the reader to familiarise with that before going forward.
+
+To open the head, we must submit an `Init`, followed by each user committing (`Commit`) some UTXOs from the Cardano main-chain to the head.
+
+Either player can send an `init n` command through the CLI, where `n` is the number of seconds allowed for contestation.
+
+After that, each participant must commit their funds. If you payed attention when running the `spin-up-devnet-from-scratch` script, you might have noticed the last lines logging something about seeding transactions.
+What happened there is that we distributed some funds to both alice and bob so that they have an UTxO to commit to the head (we also create transactions sending both of them fuel, which is needed by the node to drive the head protocol forward, fuel will be ignored for the rest of this discussion. Refer to [?](??) for more information).
```

Comment:
Will this fuel reference be added later?

Reply:
Good spot, I wanted to add a link to their docs there but forgot to add it in the end!